xtask/src/semver_check.rs

```diff
@@ -0,0 +1,278 @@
+        {
+            to_remove.push(id.clone());
+        }
+    }
+
+    log::debug!("Items to remove {:?}", to_remove);
+
+    for id in &to_remove {
+        krate.index.remove(&id);
+        krate.paths.remove(&id);
+    }
+
+    for (_id, item) in &mut krate.index {
+        match &mut item.inner {
+            ItemEnum::Module(module) => {
+                module.items = module
+                    .items
+                    .iter()
+                    .filter(|id| !to_remove.contains(id))
+                    .map(|id| id.clone())
+                    .collect();
+            }
+            ItemEnum::Struct(struct_) => {
+                struct_.impls = struct_
+                    .impls
+                    .iter()
+                    .filter(|id| !to_remove.contains(id))
+                    .map(|id| id.clone())
+                    .collect();
+
+                struct_.impls = struct_
+                    .impls
+                    .iter()
+                    .filter(|id| !to_remove.contains(id))
+                    .map(|id| id.clone())
+                    .collect();
+
+                match &mut struct_.kind {
+                    rustdoc_types::StructKind::Unit => (),
+                    rustdoc_types::StructKind::Tuple(_ids) => (),
+                    rustdoc_types::StructKind::Plain {
+                        fields,
+                        has_stripped_fields: _,
+                    } => {
+                        for id in &to_remove {
+                            if let Some(found) = fields.iter().enumerate().find(|i| i.1 == id) {
+                                fields.remove(found.0);
+                            }
+                        }
+                    }
+                }
+
+                if struct_.impls.is_empty() {
+                    to_remove.push(_id.clone());
+                }
+            }
+            ItemEnum::Enum(enum_) => {
+                enum_.impls = enum_
+                    .impls
+                    .iter()
+                    .filter(|id| !to_remove.contains(id))
+                    .map(|id| id.clone())
+                    .collect();
+
+                enum_.variants = enum_
+                    .variants
+                    .iter()
+                    .filter(|id| !to_remove.contains(id))
+                    .map(|id| id.clone())
+                    .collect();
+
+                if enum_.impls.is_empty() {
+                    to_remove.push(_id.clone());
+                }
+            }
+            ItemEnum::Impl(impl_) => {
+                impl_.items = impl_
+                    .items
+                    .iter()
+                    .filter(|id| !to_remove.contains(id))
+                    .map(|id| id.clone())
+                    .collect();
+
+                if impl_.items.is_empty() {
+                    to_remove.push(_id.clone());
+                }
+            }
+
+            // don't honor:
```

Comment:
Why aren't these honored? Because we don't current'y mark any of these items as unstable?

Reply:
Mostly because they don't have sub-items so we would have already handled them in the first-pass - but probably worth a comment / more comments 👍